### ISSUE TITLE
Orthographic projection

### ISSUE THREAD

**Issue by Yuliang-Zou** [shubhtuls/mvcSnP#2]
Hi @shubhtuls , seems that the default projection type is perspective, I wonder what should I modify so that I can get the calibration matrix of a orthographic projection camera.

**Comment by shubhtuls** [comment]
Hi @Yuliang-Zou ,
Do you mean how to modify the training code to handle orthographic cameras? Or how to change the rendering using blender to use orthographic cameras?

**Comment by Yuliang-Zou** [comment]
I mean the first one. I notice that you commented [this line](https://github.com/shubhtuls/mvcSnP/blob/master/preprocess/synthetic/rendering/renderer/render_model_views.py#L248). I think if I uncomment it, the camera type in blender is the orthographic. But I don't know how to modify the code to handle this

**Comment by shubhtuls** [comment]
To allow the loss function to have orthographic cameras, you'll need to change [this function](https://github.com/shubhtuls/mvcSnP/blob/master/loss/dprLoss.lua#L105-L125). It currently samples points along each ray and pre-determined depths, and you'll need to edit this to instead yield samples along each direction.

Also, you'll need to use an orthographic transform instead of a rigid transform [here](https://github.com/shubhtuls/mvcSnP/blob/master/loss/dprLoss.lua#L65).

I'm closing this issue because we're not planning to support orthographic cameras in the official release.

**Comment by Yuliang-Zou** [comment]
Thanks @shubhtuls .
One last question: should I modify [this function](https://github.com/shubhtuls/mvcSnP/blob/master/preprocess/synthetic/rendering/renderer/render_model_views.py#L37) if I want to use orthographic camera?

**Comment by shubhtuls** [comment]
Yes, the blender scripts will definitely change because under orthographic projections, you don't need a K matrix, just a scale, Rotation and a 2D translation to define the camera.
Unfortunately, I'm not sure how exactly things need to change in blender as I've never worked with orthographic projections in it.
